src/libraries/System.Private.CoreLib/gen/NativeRuntimeEventSourceGenerator.cs

```diff
@@ -0,0 +1,394 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+#nullable enable
+
+namespace Generators
+{
+    [Generator]
+    public sealed class NativeRuntimeEventSourceGenerator : IIncrementalGenerator
+    {
+        private static readonly XNamespace EventNs = "http://schemas.microsoft.com/win/2004/08/events";
+
+        public void Initialize(IncrementalGeneratorInitializationContext context)
+        {
+            // Register the generator to run on AdditionalFiles (manifest and inclusion list)
+            IncrementalValuesProvider<AdditionalText> manifestFiles = context.AdditionalTextsProvider.Where(f => f.Path.EndsWith(".man", StringComparison.OrdinalIgnoreCase));
+            IncrementalValuesProvider<AdditionalText> inclusionFiles = context.AdditionalTextsProvider.Where(f => f.Path.EndsWith(".lst", StringComparison.OrdinalIgnoreCase));
+
+            IncrementalValuesProvider<(AdditionalText Left, System.Collections.Immutable.ImmutableArray<AdditionalText> Right)> combined = manifestFiles.Combine(inclusionFiles.Collect());
+
+            context.RegisterSourceOutput(combined, (spc, tuple) =>
+            {
+                AdditionalText manifestFile = tuple.Left;
+                System.Collections.Immutable.ImmutableArray<AdditionalText> inclusionFiles = tuple.Right;
+                string manifestText = manifestFile.GetText(spc.CancellationToken)?.ToString();
+                string inclusionText = inclusionFiles.FirstOrDefault()?.GetText(spc.CancellationToken)?.ToString();
```

Comment:
Should we defer this until after we validate `manifestText`? Defer work when not needed.

Reply:
If this is non-null, does it mean it wil be `string.Empty`? I see a bunch of `?` and that means check for null, but `ToString()` on `null` is bad? I think a similar question exists on `manifestText`.